util/log/file.go

```diff
@@ -38,9 +39,41 @@ import (
 // MaxSize is the maximum size of a log file in bytes.
 var MaxSize uint64 = 1024 * 1024 * 10
 
-// If non-empty, overrides the choice of directory in which to write logs.
-// See createLogDirs for the full list of possible destinations.
-var logDir *string
+// If non-empty, overrides the choice of directory in which to write logs. See
+// createLogDirs for the full list of possible destinations. Note that the
+// default is to log to stderr independent of this setting. See --logtostderr.
+var logDir = os.TempDir()
+var logDirSet bool
+
+// DirSet returns true of the log directory has been changed from its default.
+func DirSet() bool {
+	return logDirSet
+}
+
+type stringValue struct {
+	val *string
+	set *bool
+}
+
+var _ flag.Value = &stringValue{}
+
+func newStringValue(val *string, set *bool) *stringValue {
+	return &stringValue{val: val, set: set}
+}
+
+func (s *stringValue) Set(val string) error {
```

Comment:
So the need to create the newStringValue is entirely for the `set`.  Why can't just detect a blank instead of adding these types?